src/bin/cargo/commands/describe_future_incompatibilities.rs

```diff
@@ -0,0 +1,50 @@
+use crate::command_prelude::*;
+use anyhow::anyhow;
+use cargo::core::compiler::future_incompat::{OnDiskReport, FUTURE_INCOMPAT_FILE};
+use cargo::core::nightly_features_allowed;
+use cargo::drop_eprint;
+use std::io::Read;
+
+pub fn cli() -> App {
+    subcommand("describe-future-incompatibilities")
+        .arg(
+            opt(
+                "id",
+                "identifier of the report [generated by a Cargo command invocation",
+            )
+            .value_name("id")
+            .required(true),
+        )
+        .about("Reports any crates which will eventually stop compiling")
+}
+
+pub fn exec(config: &mut Config, args: &ArgMatches<'_>) -> CliResult {
+    if !nightly_features_allowed() {
+        return Err(anyhow!(
+            "`cargo describe-future-incompatibilities` can only be used on the nightly channel"
+        )
+        .into());
+    }
+
+    let ws = args.workspace(config)?;
+    let report_file = ws.target_dir().open_ro(
+        FUTURE_INCOMPAT_FILE,
+        ws.config(),
+        "Future incompatible report",
+    )?;
+
+    let mut file_contents = String::new();
+    report_file
+        .file()
+        .read_to_string(&mut file_contents)
+        .map_err(|e| anyhow!("Failed to read report: {:?}", e))?;
+    let on_disk_report: OnDiskReport = serde_json::from_str(&file_contents).unwrap();
```

Comment:
This unwrap doesn't seem safe. If the on-disk format changes in the future, it could fail to load.

```suggestion
    let on_disk_report: OnDiskReport = serde_json::from_str(&file_contents)
      .chain_err(|| "failed to load report")?;
```